Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -323,7 +323,7 @@ TIFFImageIO::InitializeColors()
       itkExceptionMacro(<< "Sorry, can not handle image with " << m_InternalImage->m_BitsPerSample << "-bit samples");
   }
 
-  m_TotalColors = (1L << m_InternalImage->m_BitsPerSample);
+  m_TotalColors = static_cast<int>(1L << m_InternalImage->m_BitsPerSample);
```

Comment:
Do you know why the bit-shift was done on a `long` integer, "1L"?  This `static_cast` appears to silence a potential issue without solving it. If `int` is good enough (instead of `long`), I would think the line of code should simply look like this (without cast, and without `long` "1L"):

     m_TotalColors = (1 << m_InternalImage->m_BitsPerSample);

Reply:
The goal of static cast here is to be explicit on the unsigned -> signed casting, because  m_TotalColors is signed. I don't know why bit shift is done on a long. Code was introduced in 40afd95161727be3274f8094ce19fe45e2c3aa50 . If I had to guess, I would say for safety reason with bit shifts overhead ?

Reply:
As far as I can see, the expression `(1 << m_InternalImage->m_BitsPerSample)` is already signed. So the `static_cast<int>` does not seem necessary.  Right?

By the way, C++11 braced conversion `int{x}` is more type-safe than `static_cast<int>(x)`. Would you like that better?

    m_TotalColors = int{ 1 << m_InternalImage->m_BitsPerSample };

Reply:
In theory, int must be 16bit at least, so you could have a 16bit image, and it would be overflow. It shouldn't happen on 3ébit platform, and anyway m_TotalColors is int. Or We could set m_TotalColors to a long. What do you think ?

Reply:
On Linux, `long` is typically 64-bit, but on Windows, `int` and `long` are both 32-bits (nowadays).s. When `m_TotalColors` is a signed 32-bits integer, BitsPerSample > 30 will cause overflow. Is that OK? I mean, is BitsPerSample in practice always below 31? Otherwise maybe `m_TotalColors` should be defined as `int64_t`. Or `uint64_t`.

Reply:
I've defined it as `int64_t`, to be on the really safe side (`uint64_t` not possible because of the `-1` in case no palette)